qubes-rpc/nautilus/qvm_copy_nautilus.py

```diff
@@ -28,10 +28,30 @@ def get_file_items(self, *args):
     def on_menu_item_clicked(self, menu, files):
         '''Called when user chooses files though Nautilus context menu.
         '''
-        cmd = [file_obj.get_location().get_path()
-               for file_obj in files
-               # Check if file is not gone
-               if not file_obj.is_gone()]
+        paths = []
+        for file_obj in files:
+            file_location = file_obj.get_location()
+            file_uri = file_location.get_uri()
+            if file_uri.startswith('file:///'):
+                if not file_obj.is_gone():
+                    # Check if file is not gone
+                    paths.append(file_location.get_path())
+            elif file_uri.startswith('recent:///'):
+                try:
+                    file_info = file_location.query_info(
+                            Gio.FILE_ATTRIBUTE_STANDARD_TARGET_URI, 0, None)
+                    target_uri = file_info.get_attribute_string(
+                            Gio.FILE_ATTRIBUTE_STANDARD_TARGET_URI)
+                    if target_uri.startswith('file://'):
+                        paths.append(target_uri[7:])
+                except GLib.GError:
+                    # TODO: Decide what to do if the recent item does not exist
```

Comment:
I think that we should at least notify the user that something went wrong in those cases - and, hopefully, what went wrong. Like a "the file was moved" or something error?

Reply:
There is also another approach. To do it how it is [currently done](https://github.com/QubesOS/qubes-app-linux-img-converter/blob/main/qvm_convert_img_nautilus.py#L31-L34) in `Convert to Trusted Img` and `Convert to Trusted PDF` code.

Just hide the options if the URL is neither of  `file:///` or `recent:///`. I think it is better to remove the option from right-click menu for Network Shares for the time.

Please let me know if it is OK with you.

Reply:
Oh, I like this idea. Yeah, it's safer, I think.

Reply:
@marmarta It is done.